File: library/src/test/java/me/proxer/library/api/NotificationAdapterTest.java
package me.proxer.library.api;

import com.squareup.moshi.JsonDataException;
import me.proxer.library.entity.notifications.NotificationInfo;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

/**
 * @author Ruben Gees
 */
class NotificationAdapterTest {

    private NotificationInfoAdapter adapter;

    @BeforeEach
    void setUp() {
        adapter = new NotificationInfoAdapter();
    }

    @Test
    void testFromJson() {
        assertThat(adapter.fromJson(new int[]{0, 0, 1, 2, 3, 4}))
                .isEqualTo(new NotificationInfo(1, 2, 3, 4));
    }

    @Test
    void testFromJsonInvalidSizeTooSmall() {
        assertThatThrownBy(() -> adapter.fromJson(new int[]{1, 2, 3, 4, 5}))
                .isInstanceOf(JsonDataException.class);
    }

    @Test
    void testFromJsonInvalidSizeTooLarge() {
        assertThatThrownBy(() -> adapter.fromJson(new int[]{1, 2, 3, 4, 5, 6, 7}))
                .isInstanceOf(JsonDataException.class);
    }
}

File: library/src/test/kotlin/me/proxer/library/internal/adapter/NotificationAdapterTest.kt
package me.proxer.library.internal.adapter

import com.squareup.moshi.JsonDataException
import me.proxer.library.entity.notifications.NotificationInfo
import org.amshove.kluent.invoking
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldThrow
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class NotificationAdapterTest {

    private val adapter = NotificationInfoAdapter()

    @Test
    fun testFromJson() {
        adapter.fromJson(intArrayOf(0, 0, 1, 2, 3, 4)) shouldBeEqualTo NotificationInfo(
            messageAmount = 1,
            friendRequestAmount = 2,
            newsAmount = 3,
            notificationAmount = 4
        )
    }

    @Test
    fun testFromJsonInvalidSizeTooSmall() {
        invoking { adapter.fromJson(intArrayOf(1, 2, 3, 4, 5)) } shouldThrow JsonDataException::class
    }

    @Test
    fun testFromJsonInvalidSizeTooLarge() {
        invoking { adapter.fromJson(intArrayOf(1, 2, 3, 4, 5, 6, 7)) } shouldThrow JsonDataException::class
    }
}
